### PR TITLE
feat: Add implementation for all DID URL dereferencing parameters [DEV-2351]

### PR DESCRIPTION
**Pull request by anikitinDSR**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-2351 Add implementation for all DID URL dereferencing parameters (paths, queries, fragments) and how each should respond](https://app.clickup.com/t/6600954/DEV-2351)

File: services/diddoc/queries/resources/resource_query_collection_id_handler.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		return d.Continue(c, service, response)
+	}
+
+	// Cast to just list of resources
+	resourceCollection, err := d.CastToContent(service, response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter the list of metadatas by the resourceCollectionId
```
Comment (abdulla-ashurov) [inline services/diddoc/queries/resources/resource_query_collection_id_handler.go:26]: The noun `metadata` is uncountable. The plural form of metadata is also metadata.

File: services/diddoc/queries/resources/resource_query_type_handler.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	resourceType := service.GetQueryParam(types.ResourceType)
+	if resourceType == "" {
+		return d.Continue(c, service, response)
+	}
+
+	resourceCollection, err := d.CastToContent(service, response)
+	if err != nil {
+		return nil, err
+	}
+	// Filter the list of metadatas by the resourceCollectionId
```
Comment (abdulla-ashurov) [inline services/diddoc/queries/resources/resource_query_type_handler.go:24]: The noun `metadata` is uncountable. The plural form of metadata is also metadata.

File: services/diddoc/queries/resources/resource_query_name_handler.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		return d.Continue(c, service, response)
+	}
+
+	// Cast to just list of resources
+	resourceCollection, err := d.CastToContent(service, response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter the list of metadatas by the resourceCollectionId
```
Comment (abdulla-ashurov) [inline services/diddoc/queries/resources/resource_query_name_handler.go:27]: The noun `metadata` is uncountable. The plural form of metadata is also metadata.

File: services/diddoc/diddoc_query.go [hunk 4]
```diff
@@ -23,34 +29,283 @@ func (dd *QueryDIDDocRequestService) SpecificValidation(c services.ResolverConte
...
+	err = resourceQueryHandler.SetNext(c, &resourceIdHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// Resource handlers
+	// Chain would be:
+	// resourceIdHandler -> resourceCollectionIdHandler ->
+	// -> resourceNameHandler -> resourceTypeHandler -> resourceVersionHandler ->
+	// -> resourceVersionTimeHandler -> resourceValidationHandler -> resourceMetadataHandler -> stopHandler
```
Comment (abdulla-ashurov) [inline services/diddoc/diddoc_query.go:238]: Let's add in that comment another handlers too:
- `resourceQueryHandler`
- `resourceChecksumHandler`
- `resourceMetadataHandler`

File: services/diddoc/diddoc_query.go [hunk 5]
```diff
@@ -23,34 +29,283 @@ func (dd *QueryDIDDocRequestService) SpecificValidation(c services.ResolverConte
...
+		return types.NewRepresentationNotSupportedError(service, dd.GetContentType(), nil, dd.IsDereferencing)
+	}
+
+	// value if metadata can be only true or false
+	if metadata != "" && metadata != "true" && metadata != "false" {
+		return types.NewRepresentationNotSupportedError(metadata, dd.GetContentType(), nil, dd.IsDereferencing)
+	}
+
+	// value if resourceMetadata can be only true or false
+	if resourceMetadata != "" && resourceMetadata != "true" && resourceMetadata == "false" {
```
Comment (abdulla-ashurov) [inline services/diddoc/diddoc_query.go:71]:
```suggestion
	if resourceMetadata != "" && resourceMetadata != "true" && resourceMetadata != "false" {
```

File: services/diddoc/queries/resources/resource_helper_handler.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/cheqd/did-resolver/services"
+	"github.com/cheqd/did-resolver/types"
+)
+
+type ResourceHelperHandler struct{}
+
+func (d *ResourceHelperHandler) CastToContent(service services.RequestServiceI, response types.ResolutionResultI) (*types.ResolutionDidDocMetadata, error) {
+	// Cast to DidDocMetadataList for getting the list of metadatas
```
Comment (abdulla-ashurov) [inline services/diddoc/queries/resources/resource_helper_handler.go:11]: The noun metadata is uncountable. The plural form of metadata is also metadata.

**Review by abdulla-ashurov — COMMENTED** [review]
Great job! Thanks! 🔥🔥

**Review by abdulla-ashurov — APPROVED** [review]

**Pull request merged**